### PR TITLE
Adding TallyHo wallet to Gnosis Safe

### PR DESCRIPTION
**Pull request by bmorphism**
This adds the changes suggested by @katspaugh in https://github.com/safe-global/safe-react/pull/3741

Additionally, merged in latest state of `dev` to make the merge into your branch easier and prepares for the subsequent merge upstream by incorporating suggested edits.

### CODE REVIEW

File: src/logic/wallets/getWeb3.ts [hunk 1]
```diff
@@ -23,6 +23,7 @@ import { Wallet } from 'bnc-onboard/dist/src/interfaces'
 // This providers have direct relation with name assigned in bnc-onboard configuration
 export enum WALLET_PROVIDER {
   METAMASK = 'METAMASK',
+  TALLYHO = 'TALLYHO',
```
Comment (bmorphism) [inline src/logic/wallets/getWeb3.ts:26]: This addresses https://github.com/safe-global/safe-react/pull/3741#discussion_r838629163

**Review by beemeeupnow — APPROVED** [review]
Thank you!

**Comment by beemeeupnow** [comment]
@katspaugh FYI the comments are locked on safe-global/safe-react#3741

They need to be unlocked so that @bmorphism can sign the CLA via a comment there.

**Comment by katspaugh** [comment]
Unlocked 👌

**Pull request merged**